app/src/main/java/org/mozilla/fenix/components/Components.kt

```diff
@@ -207,7 +208,10 @@ class Components(private val context: Context) {
                     core.pocketStoriesService,
                     context.pocketStoriesSelectedCategoriesDataStore,
                 ),
-                MessagingMiddleware(messagingStorage = analytics.messagingStorage),
+                MessagingMiddleware(
+                    surface = MessageSurfaceId.HOMESCREEN,
+                    messagingStorage = analytics.messagingStorage,
+                ),
```

Comment:
I'm a bit confused about how we are going to handle multiple surface, if we are just passing one to the middleware?

Reply:
Are we just going to have one surface per app startup?

Reply:
> I'm a bit confused about how we are going to handle multiple surface

I'm not sure where the `MessagingMiddleware` fits in: does it look after one surface, or should it be looking after all surfaces? If _all_ surfaces, how does it fit in with a WorkManager job?

My thought here was that the Notification Worker would use a messaging storage directly.

Should the `MessagingSurfaceId` travel in the Action/Facts rather than the middleware directly?

Reply:
> I'm not sure where the MessagingMiddleware fits in: does it look after one surface, or should it be looking after all surfaces? If all surfaces, how does it fit in with a WorkManager job?

At the moment, it contains all the logic on handling when a message should be displayed or interacted with, ideally as all the logic practically should be the same for handling messages (Just the surface is different) we could have the middleware to take of all these actions for all surfaces and from other surfaces like the Notification Worker we could dispatch actions, indicating we would like to `Evaluate` for a specific surface the same could be done in the home screen, just by adding the surface to the `Evaluate` action. 

It will be good to define if we will be able support sending multiple messages in multiple surfaces at the same time per app startup. 


> My thought here was that the Notification Worker would use a messaging storage directly.

We could use the storage directly but we may need to duplicate some of the logic that already live on the `MessagingMiddleware`.

> Should the MessagingSurfaceId travel in the Action/Facts rather than the middleware directly?

Yeah, having the surface as part of the actions could be ideal to keep the common logic in the middleware.

Reply:
From where are we going to get the `MessageSurfaceId`? As if we get it directly from the message, we won't need to change much as the message could tell us where it should be shown,  without a lot of changes.

Reply:
> From where are we going to get the MessageSurfaceId? As if we get it directly from the message,

Yes, this is correct.

> We could use the storage directly but we may need to duplicate some of the logic that already live on the MessagingMiddleware.

Yes, that was my thought too. A judicious refactor of `MessagingMiddleware`would be required, but definitely do-able.

> It will be good to define if we will be able support sending multiple messages in multiple surfaces at the same time per app startup.

If I understand your question correctly: it's intended that multiple messages be sent to any surface; however, only one message per surface is displayed at any one time.

If there are two messages eligible for the same surface, the highest priority message is displayed. If it is clicked or dismissed by the user, or it becomes ineligible, the next highest priority message is displayed.

Reply:
Perfect, thanks for the clarification! 
I think If we add the `MessageSurfaceId` as part of the  `message`  we will need to update fewer points, only the ones that are specific about where the message should be shown :) .